backend/tests/unit_tests/config/test_read.py

```diff
@@ -58,10 +60,11 @@ def _run_r_config(
         mock_execute_query.side_effect = [schema, return_value]
 
         if expected_error:
-            with pytest.raises(ValueError) as error:
+            with pytest.raises(APIError) as error:
                 r_config(config_definition_key, config_key)
                 mock_execute_query.assert_not_called()
-            assert str(error.value) == expected_error
+            detail = error.value.detail[0]
+            assert detail["msg"] == expected_error
```

Comment:
I respectfully disagree with this suggestion. The `APIError` is supposed to be raised, in case queries are not called. This condition is right and having it outside like the suggestion might cause the test to fail.